src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts

```diff
@@ -29,21 +32,37 @@ export class YamlEventViewerComponent {
         Object.entries(this.event).forEach(([key, value]) => {
             key !== "__typename" && value ? (result += `${key}:\n`) : "";
             if (value && key !== "__typename") {
-                Object.entries(value as object).forEach(([property, data]) => {
-                    if (data) {
-                        result += `\u00A0\u00A0${
-                            property === "__typename" ? "kind" : property
-                        }: ${
-                            property === "__typename"
-                                ? this.kindMapper[data as string]
-                                : property === "queries"
-                                ? (data as SqlQueryStep[]).map(
-                                      (item) => `>\n${item.query}`,
-                                  )
-                                : (data as string)
+                if (key === "inputs") {
+                    (value as TransformInput[]).map((item: TransformInput) => {
+                        result += `\u00A0\u00A0- kind: ${
+                            this.kindMapper[item.dataset.__typename as string]
                         }\n`;
-                    }
-                });
+                        result += `\u00A0\u00A0\u00A0\u00A0id: ${item.dataset.id}\n`;
+                        result += `\u00A0\u00A0\u00A0\u00A0type: ${item.dataset.kind}\n`;
+                        result += `\u00A0\u00A0\u00A0\u00A0name: ${item.dataset.name}\n`;
+                        result += `\u00A0\u00A0\u00A0\u00A0owner: ${item.dataset.owner.name}\n`;
+                    });
+                } else {
+                    Object.entries(value as object).forEach(
+                        ([property, data]) => {
+                            if (data) {
+                                result += `\u00A0\u00A0${
+                                    property === "__typename"
+                                        ? "kind"
+                                        : property
+                                }: ${
+                                    property === "__typename"
+                                        ? this.kindMapper[data as string]
+                                        : property === "queries"
+                                        ? (data as SqlQueryStep[]).map(
+                                              (item) => `>\n${item.query}`,
+                                          )
+                                        : (data as string)
+                                }\n`;
+                            }
```

Comment:
Here we have quite a mess with data types. Sometimes single string, sometimes array of strings. I think it is better rewrite this code procedurally, with `if` and `for` statements.

Reply:
Fixed

Reply:
Refactored this peace of code.